jwst/outlier_detection/outlier_detection_step.py

```diff
@@ -177,18 +177,19 @@ def process(self, input_data):
                 for model in self.input_models:
                     model.meta.cal_step.outlier_detection = state
                     if not self.save_intermediate_results:
-                        # remove unwanted files
-                        crf_file = self.make_output_path(basepath=model.meta.filename)
-                        outlr_basename = os.path.basename(crf_file)
+                        # Remove unwanted files
+                        crf_path = self.make_output_path(basepath=model.meta.filename)
```

Comment:
So all of this file name building is being done just so that it can find the right files to delete, right? Presumably it matches the equivalent code (wherever it is) that creates/saves the files in the first place?

Reply:
Yes, and ideally the behavior would match across all files generated. Unfortunately, it looks as though the outlier_i2d files behave differently - possibly as a result of the datamodels.open(save_open) code? I'm trying to understand exactly where these files come from. The blot and median files are dumped using the typical model.save() method, and are easy to replicate.